allennlp/models/simple_tagger.py

```diff
@@ -39,23 +55,47 @@ class SimpleTagger(Model):
     def __init__(self, vocab: Vocabulary,
                  text_field_embedder: TextFieldEmbedder,
                  encoder: Seq2SeqEncoder,
+                 calculate_span_f1: bool = None,
+                 label_encoding: Optional[str] = None,
+                 label_namespace: str = "labels",
+                 verbose_metrics: bool = False,
                  initializer: InitializerApplicator = InitializerApplicator(),
                  regularizer: Optional[RegularizerApplicator] = None) -> None:
         super(SimpleTagger, self).__init__(vocab, regularizer)
 
+        self.label_namespace = label_namespace
         self.text_field_embedder = text_field_embedder
-        self.num_classes = self.vocab.get_vocab_size("labels")
+        self.num_classes = self.vocab.get_vocab_size(label_namespace)
         self.encoder = encoder
+        self._verbose_metrics = verbose_metrics
         self.tag_projection_layer = TimeDistributed(Linear(self.encoder.get_output_dim(),
                                                            self.num_classes))
 
         check_dimensions_match(text_field_embedder.get_output_dim(), encoder.get_input_dim(),
                                "text field embedding dim", "encoder input dim")
+
+        # We keep calculate_span_f1 as a constructor argument for API consistency with
+        # the CrfTagger, even it is redundant in this class
+        # (label_encoding serves the same purpose).
+        if calculate_span_f1 is None:
+            calculate_span_f1 = label_encoding is not None
+        self.label_encoding = label_encoding
+
         self.metrics = {
                 "accuracy": CategoricalAccuracy(),
                 "accuracy3": CategoricalAccuracy(top_k=3)
         }
 
+        if calculate_span_f1:
+            if not label_encoding:
+                raise ConfigurationError("calculate_span_f1 is True, but "
+                                         "no label_encoding was specified.")
+            self._f1_metric = SpanBasedF1Measure(vocab,
```

Comment:
ok, so having thought about this more, I think it would be slightly cleaner ("flat is better than nested") if you did the following. first in its own block, do

```
if calculate_span_f1 and not label_encoding:
    raise ConfigurationError(...)
```

Then in a second block do

```
if calculate_span_f1 or label_encoding:
    self._f1_metric = ...
else:
    self._f1_metric = None
```

I feel like the logic is a lot clearer than with the nested ifs